### PR TITLE
Add Global ID (and larger boundary conditions)

### PR DESCRIPTION
**Pull request by davschneller**
Adds a global ID to each element (in preparation to checkpointing). Also, we streamline 64-bit boundary conditions.

Dependent on the PUML2 PR https://github.com/TUM-I5/PUML2/pull/14 .

### CODE REVIEW

File: src/Geometry/PUMLReader.cpp [hunk 1]
```diff
@@ -60,17 +63,116 @@
...
+                   << bcToString(sideBC)
+                   << "boundary condition, but the neighboring element is not flagged -1";
+      return false;
+    }
+  }
+  // ignore unknown boundary conditions and warn
+  else {
+    logWarning() << "Element" << cellIdAsInFile << ", side" << side
+                 << " has a boundary condition, which I don't understand" << sideBC;
+    return true;
```
Comment (Thomas-Ulrich) [inline src/Geometry/PUMLReader.cpp:150]: I would also return error in this case.

File: src/Geometry/PUMLReader.cpp [hunk 2]
```diff
@@ -60,17 +63,116 @@
...
+const int PumlFaceToSeisSol[4] = {0, 1, 3, 2};
+
+const int FaceVertexToOrientation[4][4] = {
+    {0, 2, 1, -1}, {0, 1, -1, 2}, {0, -1, 2, 1}, {-1, 0, 1, 2}};
+
+const int FirstFaceVertex[4] = {0, 0, 0, 1};
+} // namespace
+
 /**
  * @todo Cleanup this code
```
Comment (Thomas-Ulrich) [inline src/Geometry/PUMLReader.cpp:164]: still required?

File: src/Geometry/PUMLReader.cpp [hunk 3]
```diff
@@ -60,17 +63,116 @@
...
+    logWarning() << "Element" << cellIdAsInFile << ", side" << side
+                 << " has a boundary condition, which I don't understand" << sideBC;
+    return true;
+  }
+  return true;
+}
+
+const int PumlFaceToSeisSol[4] = {0, 1, 3, 2};
+
+const int FaceVertexToOrientation[4][4] = {
```
Comment (Thomas-Ulrich) [inline src/Geometry/PUMLReader.cpp:157]: can you add a comment about what is orientation?

File: src/Geometry/PUMLReader.cpp [hunk 4]
```diff
@@ -60,17 +63,116 @@
...
+    {0, 2, 1, -1}, {0, 1, -1, 2}, {0, -1, 2, 1}, {-1, 0, 1, 2}};
+
+const int FirstFaceVertex[4] = {0, 0, 0, 1};
```
Comment (Thomas-Ulrich) [inline src/Geometry/PUMLReader.cpp:160]: can you add a comment about what it is?

File: src/Geometry/PUMLReader.h [hunk 5]
```diff
@@ -41,28 +41,48 @@
 
 #include "MeshReader.h"
 #include "Parallel/MPI.h"
-#include "PUML/PUML.h"
+#include <Initializer/Parameters/MeshParameters.h>
```
Comment (Thomas-Ulrich) [inline src/Geometry/PUMLReader.h:44]: shouldn't it be "Initializer/Parameters/MeshParameters.h"?

**Review by Thomas-Ulrich — CHANGES_REQUESTED** [review]

**Comment by davschneller** [comment]
Ok; I've added an `auto` option which checks the HDF5 file. (if we ever should add bare-binary Xdmf support or Hdf5 data from different files, we may need to read the XML file instead)

**Comment by palgunadi1993** [comment]
I tested this PR. It works fine with my fracture network model. Thanks @davschneller !

**Review by Thomas-Ulrich — APPROVED** [review]

**Pull request merged**